File: CardMasterSkin/Skins/Skin.vb
Imports System.Drawing
Imports System.IO

Namespace Skins

    Public Class Skin

        Public Property Width As Integer
        Public Property Height As Integer
        Public Property ImagesDirectory As DirectoryInfo
        Public Property TexturesDirectory As DirectoryInfo
        Public Property Elements As New SkinElements


        Protected Sub New()
        End Sub

        Public Sub New(width As Integer, height As Integer, imagesDirectory As DirectoryInfo, texturesDirectory As DirectoryInfo)

            Me.Width = width
            Me.Height = height
            Me.ImagesDirectory = imagesDirectory
            Me.TexturesDirectory = texturesDirectory

        End Sub

        Protected Overrides Sub Finalize()

            Me.ImagesDirectory = Nothing
            Me.TexturesDirectory = Nothing
            Me.Elements.Clear()

            MyBase.Finalize()
        End Sub
    End Class

End Namespace

File: CardMasterSkin/Skins/Skin.cs
using System.IO;

namespace CardMasterSkin.Skins
{
    public class Skin
    {
        public int Width { get; set; }
        public int Height { get; set; }
        public DirectoryInfo ResourcesDirectory { get; set; }
        public SkinElements Elements { get; set; }


        protected Skin()
        { }

        public Skin(int width, int height, DirectoryInfo resourcesDirectory)
        {
            this.Width = width;
            this.Height = height;
            this.ResourcesDirectory = resourcesDirectory;
            this.Elements = new SkinElements();
        }

        ~Skin()
        {
            this.ResourcesDirectory = null;
            this.Elements.Clear();
        }

    }
}
